components/compliance-service/api/reporting/server/server.go

```diff
@@ -337,6 +339,34 @@ func jsonExport(stream reporting.ReportingService_ExportServer) exportHandler {
 	}
 }
 
+func xmlExport(stream reporting.ReportingService_ExportServer) exportHandler {
+	initialRun := true
+	return func(data *reporting.Report) error {
+		raw, err := util.ReportToXML(data)
+		if err != nil {
+			return fmt.Errorf("Failed to marshal XML export data: %+v", err)
+		}
+
+		if initialRun {
+			initialRun = false
+		} else {
+			raw = append([]byte(","), raw...)
```

Comment:
I don't think this is what we want.. when we are exporting more than one node, this will create a comma separated set of xml, one for each node.  Instead, I think we should wrap the set with <nodes></nodes> and lose the ','

Reply:
Yeah, thanks for pointing it out, I am keeping this to `raw = append([]byte("\n"), raw...)`

so the output will be:

```
  <Node>
    <ID>1</ID>
  </Node>
  <Node>
    <ID>2</ID>
  </Node>
```

 Wrapping with `<Nodes></Nodes>` make sense to me

```
       if initialRun {
            raw = append([]byte("<Nodes>\n"), raw...)
			initialRun = false
		} else {
			raw = append([]byte("\n"), raw...)
		}
		reader := bytes.NewReader(raw)


```


But not found a suitable way to add `\n</Nodes>` at the end of the byte string any pointers? Thanks!

Reply:
>>But not found a suitable way to add \n</Nodes> at the end of the byte string any pointers? Thanks!
Good question.. let me think about that one a bit

Reply:
Treating XML as if it was just strings like this is quite brittle. Would much prefer that you gather the collection of individual `<Node>` items together, then output `<Nodes>` + collection-of-node-items + `</Nodes>` at one time.

Reply:
> Treating XML as if it was just strings like this is quite brittle. Would much prefer that you gather the collection of individual `<Node>` items together, then output `<Nodes>` + collection-of-node-items + `</Nodes>` at one time. That is, always emit valid XML elements. (Assuming it is possible in this case.)

not really possible here @msorens... we're streaming it

Reply:
This code should return a definitive number of IDs which we are looping through:
```
reportIDs, err := srv.es.GetReportIds(esIndex, formattedFilters)
```

Probably here:

```
total := len(reportIDs.Ids)
// Step 2: get all reports one by one in reverse order to be sorted asc by end_time
for idx := total - 1; idx >= 0; idx-- {
```

We may utilize this idx to identify the first and the last nodes and indicate the exporter.

Reply:
I see how to fix this.. working on it.. will be up very soon